src/backend/storeManagers/hyperplay/utils.ts

```diff
@@ -419,3 +419,102 @@ export const runModPatcher = async (appName: string) => {
     throw new Error(`Error running patcher: ${error}`)
   }
 }
+
+type SafeRemoveDirectoryOptions = {
+  maxRetries?: number
+  retryDelay?: number
+  sizeThresholdMB?: number
+}
+
+/**
+ * Safely removes a directory with retry logic to handle Windows EPERM issues
+ * @param directory Path to the directory to remove
+ * @param options Optional configuration for removal
+ * @param options.maxRetries Maximum number of removal attempts before giving up (default: 5)
+ * @param options.retryDelay Delay in milliseconds between removal attempts (default: 10000)
+ * @param options.sizeThresholdMB Threshold in MB above which async removal is used (default: 500)
+ * @returns True if directory was successfully removed or doesn't exist, false otherwise
+ * @warning This function MUST always be awaited to prevent race conditions
+ */
+export async function safeRemoveDirectory(
+  directory: string,
+  {
+    maxRetries = 10,
+    retryDelay = 6000,
+    sizeThresholdMB = 500
+  }: SafeRemoveDirectoryOptions = {}
+): Promise<boolean> {
+  if (!existsSync(directory)) {
+    return true // Directory doesn't exist, nothing to remove
+  }
+
+  // Log start of removal process
+  logInfo(`Starting removal of directory ${directory}`, LogPrefix.HyperPlay)
+
+  // Import fs promises for async operations only when needed
+  const fsPromises = await import('fs/promises')
+
+  for (let attempt = 1; attempt <= maxRetries; attempt++) {
+    try {
+      // Use different removal strategies based on expected size
+      // For directories larger than threshold, use async removal to not block the main thread
+      if (sizeThresholdMB > 250) {
+        try {
+          await fsPromises.rm(directory, {
+            recursive: true,
+            force: true,
+            maxRetries: 3
+          })
+        } catch (asyncError) {
+          // Fallback to sync if async fails
+          logWarning(
+            `Async removal failed, falling back to sync removal: ${asyncError}`,
+            LogPrefix.HyperPlay
+          )
+          rmSync(directory, { recursive: true, force: true, maxRetries: 3 })
+        }
+      } else {
+        // For smaller directories, use sync removal
+        rmSync(directory, { recursive: true, force: true, maxRetries: 3 })
+      }
+
+      // Verify directory was actually removed
+      try {
+        await fsPromises.access(directory)
+        // If we get here, directory still exists
+        logWarning(
+          `Failed to remove directory ${directory} on attempt ${attempt}/${maxRetries}, directory still exists`,
+          LogPrefix.HyperPlay
+        )
```

Comment:
do you think we should track this as a metric for observability?

Reply:
Good point. I will try to include this  into the new track event.